tests/test_class.py

```diff
@@ -2,47 +2,85 @@
+
+        """
+            
+        assert (jax_has_gpu() == 'gpu') #check that gpu works properly
+        reference_traj = np.load('files/dancing_spheres_ref.npy') #load reference trajectory
+
+        #test SD
+        traj, _, _, _ = main.main(
             1000, 10, 0.05, 50, 50, 50, 3, 0.5,
             0., 1, 0.5, 0.001, 0., 1, 1.,
             jnp.array([[-5., 0., 0.], [0., 0., 0.], [7., 0., 0.]]),
             0, 0, 0, 0, 0., 0.,
-            'None', 0, 0, 0,np.array([0]), np.array([0]) )
+            'None', 0, 0, 0,np.array([0]), np.array([0]),
+            2,0,
+            0.1,0.1)
         error = np.linalg.norm(reference_traj-traj)
         assert (error < 1e-8)
+        
+        #test RPY
+        traj, _, _, _ = main.main(
+            1000, 10, 0.05, 50, 50, 50, 3, 0.5,
+            0., 1, 0.5, 0.001, 0., 1, 1.,
+            jnp.array([[-5., 0., 0.], [0., 0., 0.], [7., 0., 0.]]),
+            0, 0, 0, 0, 0., 0.,
+            'None', 0, 0, 0,np.array([0]), np.array([0]),
+            1,0,0,0)
+        error = (np.linalg.norm(reference_traj-traj)) / np.linalg.norm(reference_traj)
+        assert (error < 0.02)
 
     def test_external_shear(self):
-    
-        assert (jax_has_gpu() == 'gpu')
-    
-        reference_traj = np.load('files/shear_pair_ref.npy')
-        dr = 0.01
-        traj, _, _ = main.main(
+        """Physical unit test for deterministic shear part of hydrodynamic calculations. Test pair of particle in simple shear against analytical result.
+
+        """
+        
+        assert (jax_has_gpu() == 'gpu') #check that gpu works properly
+        reference_traj = np.load('files/shear_pair_ref.npy') #load reference trajectory
+        dr = 0.01 #set initial surface-to-surface distance of pair of particle in simple shear
+        
+        #test SD
+        traj, _, _, _ = main.main(
             1000, 10, 0.01, 50, 50, 50, 2, 0.5,
             0., 1, 0.5, 0.001, 0., 0, 1.,
             jnp.array([[0., 1.+dr, 0.], [0., -1.-dr, 0.]]),
             0, 0, 0, 0, 0.1, 0.,
-            'None', 0, 0, 0,np.array([0]), np.array([0]) )
+            'None', 0, 0, 0,np.array([0]), np.array([0]),
+            2,0,0,0)
         error = np.linalg.norm(reference_traj-traj)
         assert (error < 1e-8)
+        
+        #test RPY
+        traj, _, _, _ = main.main(
+            1000, 10, 0.01, 50, 50, 50, 2, 0.5,
+            0., 1, 0.5, 0.001, 0., 0, 1.,
+            jnp.array([[0., 1.+dr, 0.], [0., -1.-dr, 0.]]),
+            0, 0, 0, 0, 0.1, 0.,
+            'None', 0, 0, 0,np.array([0]), np.array([0]),
+            1,0,0,0)
+        error = np.linalg.norm(reference_traj-traj)/np.linalg.norm(reference_traj)
+        assert (error < 0.17)
 
     def test_thermal_1body(self):
+        """Physical unit test for non-deterministic part of hydrodynamic calculations. Test Brownian motion of a single particle, in periodic boundary conditions.
```

Comment:
Same as comment below, split the docstring.

Reply:
Fixed

Reply:
Added type and array dimension in description of each function